Index: misc/Makefile.in
===================================================================
--- misc/Makefile.in	(revision 280)
+++ misc/Makefile.in	(working copy)
@@ -32,12 +32,13 @@
 transform       = @program_transform_name@
 LDPROG		= $(shell echo ld | sed '$(transform)')
 LD64PROG	= $(shell echo ld64 | sed '$(transform)')
+LIPO            = $(shell echo lipo | sed '$(transform)')
 
 MYINCLUDES      = -I$(builddir) -I$(srcdir) -I$(top_builddir)/include \
 			-I$(top_srcdir)/include $(INCPRIVEXT) $(FORHDRS)
 MYWARNINGS      = 
 MYDEFS          = -DBINDIR="\"$(bindir)\"" -DLDPROG="\"$(LDPROG)\"" \
-			-DLD64PROG="\"$(LD64PROG)\""
+			-DLD64PROG="\"$(LD64PROG)\"" -DLIPO="\"$(LIPO)\""
 MYLDFLAGS	= -L$(top_builddir)/libstuff
 MYLIBS		= -lstuff
 
Index: misc/libtool.c
===================================================================
--- misc/libtool.c	(revision 280)
+++ misc/libtool.c	(working copy)
@@ -2962,7 +2962,7 @@
 	if(narchs > 1){
 	    tellProjectBuilder("Combining into %s", "", cmd_flags.output);
 	    reset_execute_list();
-	    add_execute_list("lipo");
+	    add_execute_list(LIPO);
 	    add_execute_list("-create");
 	    add_execute_list("-output");
 	    add_execute_list(cmd_flags.output);
